Error grabcut.cpp: Line 178

Hi, I have a problem with a grabcut error. I have test it in 3.4.3/4 and 4.0. 
Sometimes the function throw an Error. I don't see the trigger.

I have add 2 images with the mask to reproduce
[images.tar.gz](https://github.com/opencv/opencv/files/2619607/images.tar.gz)

###### supplement
I have forgot the most important thing. Is its only if the the mask don't have the cv.GC_BGD "layer"
I have test it again with 3.4.1 and 3.4.2, it works correct


##### Code

```
bgdModel = np.zeros((1,65),np.float64)
fgdModel = np.zeros((1,65),np.float64)
cv2.imwrite("/TEST/mask.png", mask)
cv2.imwrite("/TEST/org_img.png", org_img)
mask, bgdModel, fgdModel = cv2.grabCut(org_img,mask,None,bgdModel,fgdModel,3,cv2.GC_INIT_WITH_MASK) # 6
```

##### ERROR
```
mask, bgdModel, fgdModel = cv2.grabCut(org_img,mask,None,bgdModel,fgdModel,3,cv2.GC_INIT_WITH_MASK) # 6
cv2.error: OpenCV(4.0.0) /opencv-4.0.0/modules/imgproc/src/grabcut.cpp:178: error: (-215:Assertion failed) totCount > 0 in function \'endLearning\'
```



##### System information (version)

- OpenCV => 3.4.3, 3.4.4, 4.0.0
- Operating System / Platform => Docker (FROM python:3.6)
- Compiler => make build with python3
```
FROM python:3.6

ENV OPENCV=4.0.0
ENV CPU_BASELINE=AVX2
ENV PCOUNT=40

RUN pip install --upgrade pip numpy requests Pillow logaugment
RUN buildDeps=' \
		build-essential \
		cmake \
        wget \
        unzip \
        yasm \
        git \
	' \
    && apt-get update \
    && apt-get upgrade -y \
    && apt-get install -y \
            $buildDeps \
            g++ \
            pkg-config \
            libswscale-dev \
            libtbb2 \
            libtbb-dev \
            libjpeg-dev \
            libpng-dev \
            libtiff-dev \
            libavformat-dev \
            libpq-dev \
            libcairo2-dev \
            libpango1.0-dev \
            libgif-dev \
    && wget https://github.com/opencv/opencv/archive/$OPENCV.zip \
    && unzip ${OPENCV}.zip
    RUN mkdir /opencv-${OPENCV}/cmake_binary \
    && cd /opencv-${OPENCV}/cmake_binary \
    && cmake -DBUILD_TIFF=ON \
      -DBUILD_opencv_java=OFF \
      -DCPU_BASELINE=${CPU_BASELINE} \
      -DBUILD_TESTS=OFF \
      -DBUILD_PERF_TESTS=OFF \
      -DCMAKE_BUILD_TYPE=RELEASE \
      -DCMAKE_INSTALL_PREFIX=$(python3.6 -c "import sys; print(sys.prefix)") \
      -DPYTHON_EXECUTABLE=$(which python3.6) \
      -DPYTHON_INCLUDE_DIR=$(python3.6 -c "from distutils.sysconfig import get_python_inc; print(get_python_inc())") \
      -DPYTHON_PACKAGES_PATH=$(python3.6 -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())") .. \
    && make -j$PCOUNT \
    && make install \
    && rm /${OPENCV}.zip \
    && rm -r /opencv-${OPENCV} \
    && cd $(python3.6 -c "import sys; print(sys.prefix)")/python \
    && pip install -e .
```

--- a/modules/imgproc/src/grabcut.cpp
+++ b/modules/imgproc/src/grabcut.cpp
@@ -174,7 +174,6 @@ void GMM::addSample( int ci, const Vec3d color )
 
 void GMM::endLearning()
 {
-    CV_Assert(totalSampleCount > 0);
     for( int ci = 0; ci < componentsCount; ci++ )
     {
         int n = sampleCounts[ci];
@@ -182,6 +181,7 @@ void GMM::endLearning()
             coefs[ci] = 0;
         else
         {
+            CV_Assert(totalSampleCount > 0);
             double inv_n = 1.0 / n;
             coefs[ci] = (double)n/totalSampleCount;
 

